### PR TITLE
Changed logging to use logfmt

### PR DESCRIPTION
**Pull request by doriskwan**
Added logfmt as a dependency in the pom file.

Changed defaultLoggingFormatter to use logfmt.

Adjusted test to match the new format.

For the Java logging inconsistencies [GUS](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000PEuqYAG/view)

Fixes GUS-W-9646993

### CODE REVIEW

**Comment by schneems** [comment]
I took a look. It seems good on the surface. I'll wait on @Malax to give the final approval though. Great work!

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:38]: I think fields that contain a space need to be quoted. Otherwise consumers have a hard time figuring out where a value with spaces starts and where it ends. Examples from https://brandur.org/logfmt support this. I couldn't find a spec to point to though.

When we implement this, I'd love to see a test-case for a message with a `"` inside it. I don't know how `logfmt` escapes, but we need to make sure we do it correctly.

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java:46]: Not an issue introduced in this PR, but something we need to fix regardless (also in Node.js): using a local date time string will make it harder to debug issues. The language runtimes are used in a highly distributed solution. I suggest we move this to UTC (ideally ISO 8601) and change the `logfmt` key to something that reflects this.

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java:37]: Reading the source for [logfmt-java](https://github.com/naaman/logfmt-java/blob/master/src/main/java/com/heroku/logfmt/Logfmt.java), this does note make sense to me. `Logfmt#parse` parses a `logfmt` formatted line and splits the different key/value pairs, putting them in the resulting `Map<String, char[]>`.

The code here builds a pseudo `logfmt` line, parses it with the library and then puts it together again. I'm not sure why it does this, but as-is, this doesn't do anything to ensure the log line is properly formatted (quoting, escaping).

From what I've seen, the linked `logfmt-java` library only does parsing, not generating.

**Review by Malax — CHANGES_REQUESTED** [review]

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (Malax) [reply sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:38]: The old test tested for a newline at the end, the new one doesn't. Do we know why? Are our `logfmt` formatted lines separated with a newline?

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (doriskwan) [reply sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java:46]: @Malax should we convert it before we store it the map which logfmt parses into?

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (doriskwan) [reply sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:38]: @Malax yeah, I can definitely add quotes for the message " ", but I'll definitely have to see if it works with the logfmt in java, I think there's not that much functionality in the java version vs go/node. 

it doesn't have a \n but I could definitely add it into the end of the stringbuilder

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:42]: The `message` field is still not quoted and the appropriate test for required tests inside the quotes is missing. This applies to all other tests as well.

**Review by Malax — CHANGES_REQUESTED** [review]
Thanks for improving the PR @doriskwan! Happy to talk you through the remaining issues today if things are not 💯  clear.

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (doriskwan) [reply sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java]: Added the clock back in, and got the timezone (clock.getZone()), and used that as a field instead of 'localDateTime'

Comment (doriskwan) [reply sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java]: Removed the localDateTime and used clock again and got the ZoneId associated with the clock.

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (doriskwan) [reply sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java]: Removed logfmt all together, with the dependencies and based my new PR on the expectations of logfmt with the quotations

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:86]: This is incorrect. The expected result would be:
```
"\"UTC\"=\"00:00:00.000\" \"level\"=\"INFO\" \"loggerName\"=\"blank\" \"message\"=\"Checking Quotations \\\"test\\\"\" \"invocationId\"=\"e3a4ae2b-fefb-4277-89d0-7068e7e39b99\" \n"))));
```
Since this is a String literal, all escapes are for the literal, not the actual string. That's why we'd need to double escape the backslashes and quotes for around `"test"`. 

When this is fixed, this will show that quoting is not implemented yet.

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java:32]: I think this is a misunderstanding. I think we should just use ISO 8601 with a TZ identifier ([ISO_OFFSET_DATE_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_OFFSET_DATE_TIME)) for the datetime string and use a static key for it in the log message. 

If we use a dynamic key, grepping logs would become very hard since the key might differ between machines. Also, even when in the same timezone (as in offset to UTC), a different zone id might be returned. It could be a geographical id like `Europe/Paris` or `Europe/Berlin` (both referring to CET/CEST depending on DST).

File: sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/main/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatter.java:48]: This does not do any escaping. For details, see my comment below for the test that is supposed to test this. Please note that we also need to escape the escape character `\`.

File: sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java
Comment (Malax) [inline sf-fx-runtime-java-logger/src/test/java/com/salesforce/functions/jvm/runtime/logger/DefaultLoggingFormatterTest.java:24]: This looks like it's unused.

**Review by Malax — CHANGES_REQUESTED** [review]

**Comment by Malax** [comment]
@doriskwan I went ahead and implemented quoting and escaping based on [go-logfmt](https://github.com/go-logfmt/logfmt) so we can finish this up before the weekend.

I found some other issues while going though the code in more detail and fixed them as well. I tried to have clean commits so you can follow what I did in which order. Let's meet up later to walk through the changes.

**Review by Malax — APPROVED** [review]

**Review by edmorley — APPROVED** [review]

**Pull request merged**
